### PR TITLE
[GAIA Compiler] Build algebra layer structures for graph operators

### PR DESCRIPTION
**Pull request by shirly121**
## What do these changes do?
1. build algebra layer structures for graph operators : Source/Expand/GetV/PathExpand/Match
2. introduce interfaces and implementations for fetching schema (under package of `com.alibaba.graphscope.common.ir.schema`)
3. introduce data types to denote vertex or edge (under package of `com.alibaba.graphscope.common.ir.type`)

## Related issue number

Fixes

### CODE REVIEW

File: interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/ExpandTest.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+                        .expand(
+                                new ExpandConfig(
+                                        GraphOpt.Expand.OUT,
+                                        new LabelConfig(false).addLabel("knows")))
+                        .getV(
+                                new GetVConfig(
+                                        GraphOpt.GetV.END,
+                                        new LabelConfig(false).addLabel("person")))
+                        .range(1, 3)
+                        .pathOpt(PathOpt.Simple)
```
Comment (longbinlai) [inline interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/ExpandTest.java:90]: PathOpt和ResultOpt可否也统一到GraphOpt.PathExpand.Simple和 GraphOpt.PathExpand.AllV ?

File: interactive_engine/compiler/src/main/java/com/alibaba/graphscope/common/ir/rel/type/TableConfig.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+public class TableConfig {
+    // indicate whether all labels are requested, i.e. g.V()
+    private boolean isAll;
+    // we have derived the table type(s) for the query given label(s), should not be empty
+    private List<RelOptTable> tables;
+
+    public TableConfig(List<RelOptTable> tables) {
+        this.tables = ObjectUtils.requireNonEmpty(tables);
+        ;
+        this.isAll = false;
```
Comment (longbinlai) [inline interactive_engine/compiler/src/main/java/com/alibaba/graphscope/common/ir/rel/type/TableConfig.java:38]: Please fix code format issue.

File: interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/ExpandTest.java (hunk 1 above)
Comment (shirly121) [reply interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/ExpandTest.java:90]: PathOpt和ResultOpt还是要区分开，GraphOpt里加了PathExpandPath和PathExpandResult

File: interactive_engine/compiler/src/main/java/com/alibaba/graphscope/common/ir/rel/type/TableConfig.java (hunk 2 above)
Comment (shirly121) [reply interactive_engine/compiler/src/main/java/com/alibaba/graphscope/common/ir/rel/type/TableConfig.java:38]: done

**Review by longbinlai — APPROVED** [review]

**Pull request merged**